pyracf/scripts/install_precheck.py

```diff
@@ -0,0 +1,34 @@
+from pyracf import ResourceAdmin, SecurityRequestError
+
+
+def define_precheck_profile():
+    resource_admin = ResourceAdmin()
+    try:
+        access = resource_admin.get_my_access("IRR.IRRSMO00.PRECHECK", "XFACILIT")
+    except SecurityRequestError:
+        traits_precheck = {"base:universal_access": "None"}
+        result = resource_admin.add(
+            "IRR.IRRSMO00.PRECHECK", "XFACILIT", traits=traits_precheck
+        )
+        print(
+            "IRR.IRRSMO00.PRECHECK is now defined with a `Universal Access` of None."
+            + "\nContact your security administrator for READ access before using pyRACF."
+            + "\nOther users of pyRACF will also need to have at least read access."
+            + "\nYou may also need to REFRESH the `XFACILIT` class."
+            + "\nReview our documentation at https://ambitus.github.io/pyracf/ as well!"
+        )
+        return result
+    if access:
+        print(
+            f"IRR.IRRSMO00.PRECHECK is already defined, and you already have {access} access!"
+            + "\nYou are ready to start using pyRACF!"
+            + "\nPlease ensure other users of pyRACF also have at least read access."
+            + "\nReview our documentation at https://ambitus.github.io/pyracf/ as well!"
+        )
+        return True
+    print(
+        "IRR.IRRSMO00.PRECHECK is already defined, but you have no access."
+        + "\nContact your security administrator for READ access before using pyRACF."
+        + "\nReview our documentation at https://ambitus.github.io/pyracf/ as well!"
+    )
+    return False
```

Comment:
Just to confirm that I understand how this works. This code does the following:

1. get the user's current access to `IRR.IRRSMO00.PRECHECK`.
2. If `SecurityRequestError` is raised _(because precheck isn't set up?)_, try to define precheck with universal access of `None`.
3. If the user has access, they can use precheck.
4. If precheck is setup but the user doesn't have access, they have to contact a system programmer/security administrator.
5. My guess is that if adding the precheck profile fails, it is just supposed to let the `SecurityRequestError` bubble up.

Reply:
That is correct, there's no explicit handling of any errors in the script except the initial error on step 1 that leads into step 2. If step 1 fails for any other reason (only thing i can think of is they lack access to view information on the profile), then step 2 will as well and surfacing that error is fine.

If the user has any non-NONE access then they have at least READ authority, so that is why the check says they can use precheck. Defining the profile also does mention that an admin should be contacted as well.